mne_bids/read.py

```diff
@@ -200,26 +200,39 @@ def _handle_electrodes_reading(electrodes_fname, coord_frame, raw, verbose):
     if verbose:
         print("The read in electrodes file is: \n", electrodes_dict)
 
+    def _float_or_nan(val):
+        if val == "n/a":
+            return np.nan
+        else:
+            return float(val)
+
     # convert coordinates to float and create list of tuples
-    ch_names_raw = [x for i, x in enumerate(ch_names_raw)
-                    if electrodes_dict['x'][i] != "n/a"]
-    electrodes_dict['x'] = [float(x) for x in electrodes_dict['x']
-                            if x != "n/a"]
-    electrodes_dict['y'] = [float(x) for x in electrodes_dict['y']
-                            if x != "n/a"]
-    electrodes_dict['z'] = [float(x) for x in electrodes_dict['z']
-                            if x != "n/a"]
+    electrodes_dict['x'] = [_float_or_nan(x) for x in electrodes_dict['x']]
+    electrodes_dict['y'] = [_float_or_nan(x) for x in electrodes_dict['y']]
+    electrodes_dict['z'] = [_float_or_nan(x) for x in electrodes_dict['z']]
 
     ch_locs = np.array(list(zip(electrodes_dict['x'],
-                                electrodes_dict['y'],
-                                electrodes_dict['z'])))
-    ch_pos = dict(zip(ch_names_raw, ch_locs))
+                       electrodes_dict['y'],
+                       electrodes_dict['z'])))
+
+    # determine if there are problematic channels
+    problematic_chs = []
+    for ch_name, ch_coord in zip(ch_names_raw, ch_locs):
+        if any(np.isnan(ch_coord)) and ch_name not in raw.info['bads']:
+            problematic_chs.append(ch_name)
+    if problematic_chs != []:
+        warn("Not setting montage... "
+             "There is a list of problematic channels "
+             "that are not set as 'bad', "
+             "but also the channel location "
+             "is np.nan: {}".format(problematic_chs))
+        return raw
```

Comment:
I would warn but still set whatever is available in electrodes.tsv

Reply:
@jasmainak suggested: "I wouldn't do magic. Raise a warning with the list of channels which show this problem -- and don't set the channel locations if that's the case."

I'm okay with either case as long as `read_raw_bids` doesn't cause an error in reading if there is some minor error in the electrodes.tsv file.

Reply:
The reason I suggested not setting a montage is because I thought functions downstream in MNE such as `raw.plot_psd` which do need the channel locations will work if all of them are present or none.